### DISCUSSION TITLE
Creating new GameplayTags

### DISCUSSION THREAD

**Discussion by Foffy** [Sixze/ALS-Refactored#271] [Q&A]
Apologies - I am still learning UE :)

I am trying to create my own GameplayTags, but I keep getting an "unresolved external symbol" issue when compiling.
"GameplayTags" exists in my build.cs file.

An example of the error I am getting:

`LNK2001: unresolved external symbol "class FNativeGameplayTag WestItemTypeTags::MeleeWeapon" (?MeleeWeapon@WestItemTypeTags@@3VFNativeGameplayTag@@A)`

What I did so far was to create a new .h file where I put my new tags in like this:

```
namespace WestItemTypeTags
{
	WEST_API UE_DECLARE_GAMEPLAY_TAG_EXTERN(MeleeWeapon)
	WEST_API UE_DECLARE_GAMEPLAY_TAG_EXTERN(RangedWeapon)
	WEST_API UE_DECLARE_GAMEPLAY_TAG_EXTERN(HeadArmor)
	WEST_API UE_DECLARE_GAMEPLAY_TAG_EXTERN(ChestArmor)
	WEST_API UE_DECLARE_GAMEPLAY_TAG_EXTERN(LegsArmor)

}
```

Do I need to somehow register these another place first?

**Comment by Foffy** [comment]
I... I realize now that I had made quite a brain fart here.

After a cup of coffee and a nice walk outside, I remembered that I never created a corresponding .cpp file for it!

Just in case anyone else are doing the same mistake, I'll keep this here to remind them :)

**Comment by Sixze** [comment]
You can add them via project settings: https://docs.unrealengine.com/4.26/en-US/ProgrammingAndScripting/Tags/

**Reply by Foffy** [reply]
Thank you for this comment as well. I do prefer to have the tags in my "own" files, so I can keep my project more organized, but this could definitely also work just fine! Perhaps it is even the better solution for most people!
